test/updater_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+#include <string>
+#include <vector>
+
+namespace google {
+
+class UpdaterTest : public ::testing::Test {
+ protected:
+  // query_metadata function not needed to test callbacks.
+  UpdaterTest() : config(), store(config), updater(
+      config, &store, "Test", 60.0, nullptr) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() const {
+    return store.GetMetadataMap();
+  }
+
+  bool ValidateConfiguration() const {
+    return updater.ValidateConfiguration();
+  }
+
+  void UpdateMetadataCallback(MetadataUpdater::ResourceMetadata& result) {
+    updater.UpdateMetadataCallback(std::move(result));
+  }
+
+  void UpdateResourceCallback(MetadataUpdater::ResourceMetadata& result) {
+    updater.UpdateResourceCallback(std::move(result));
```

Comment:
No need to `std::move` — `UpdateResourceCallback` takes a const reference.

Reply:
Done.